kernel/src/scan/log_replay.rs

```diff
@@ -37,80 +37,108 @@ struct LogReplayScanner {
     seen: HashSet<FileActionKey>,
 }
 
-#[derive(Default)]
-struct AddRemoveVisitor {
-    adds: Vec<(Add, usize)>,
-    removes: Vec<Remove>,
-    selection_vector: Option<Vec<bool>>,
-    // whether or not we are visiting commit json (=true) or checkpoint (=false)
+struct AddRemoveDedupVisitor<'seen> {
```

Comment:
nit: maybe quick docs sorry for asking for docs so much lol and probably useful to derive debug? (i think every field already implements Debug?)
```suggestion
#[derive(Debug)]
struct AddRemoveDedupVisitor<'seen> {
```

Reply:
Doc comment added. Is it normal in rust to derive Debug for non-public classes that don't (yet) need it?

Reply:
we can leave it without - generally i `[derive(Debug)]` on nearly everything just under the assumption that someone will want to debug print it or log in the future etc.